src/main.c

```diff
@@ -2008,7 +2008,12 @@ static gboolean on_permission_request(WebKitWebView *webview,
     char *msg = NULL;
 
     if (WEBKIT_IS_GEOLOCATION_PERMISSION_REQUEST(request)) {
-        msg = "request your location";
+        Setting* location_setting = (Setting*)g_hash_table_lookup(c->config.settings, "location");
+        if (location_setting && location_setting->value.b) {
```

Comment:
You could use the macro GET_BOOL(c, "location") to get the configured flag. But I thins it would be better to turn the "location" into "geolocation" and to allow three values like "allow",  "deny" and "ask" or something similar. So the user can allow or deny all geolocation requests or answers each request separate.

Reply:
OK, I have updated it to work like that.